### DISCUSSION TITLE
Revocation status check of peer certificates.

### DISCUSSION THREAD

**Discussion by sriramec** [strongswan/strongswan#1952] [General]
Does strongswan check the revocation status of peer certificate. Lets say in host-to-site mode, certificates are exchanged in IKE_AUTH packets and after that does strongswan client checks the revocation status of server certificate ?

Regards,
Sriram

**Comment by tobiasbrunner** [comment]
Yes, if the _revocation_ plugin is loaded and a fetcher plugin (e.g. _curl_) is available (although CRLs can also be provided locally). The `revocation` option in [swanctl.conf](https://docs.strongswan.org/docs/5.9/swanctl/swanctlConf.html) controls how the status is enforced.

**Comment by sriramec** [comment]
Thanks @tobiasbrunner. I will try that out. one more related question, does this plugin help check the revocation status of local certificate also ?

**Reply by tobiasbrunner** [reply]
No, local certificates are ultimately trusted.

**Reply by sriramec** [reply]
Thanks Tobias. 
How does the strongswan client does the signature verification of OCSP response related to peer certificate . Is it mandatory that, the signature is generated by the CA who issued the peer certificate. 
I was going through this ticket which says that CRL's authority key identifier should match with CA's subject key identifier. 
https://wiki.strongswan.org/issues/2786 
Does it holds good for OCSP also ?

**Comment by strongX509** [comment]
No it's not mandatory that the CA itself has do sign the OCSP response. The CA can delegate this to an OCSP signer whose certificate is signed by the CA and contains an _OCSPSigning_ Extended Key Usage (EKU) extension.

**Reply by sriramec** [reply]
ok Andreas. Thanks for the explanation.

**Reply by sriramec** [reply]
Hi,

Is there a way to control ocsp (relaxed, ifuri, strict) policy through ipsec.conf ? I m configuring strongswan client in host-to-site mode using ipsec.conf, so wanted to know if this is possible through ipsec.conf

Regards,
Sriram

**Reply by strongX509** [reply]
With **`swanctl.conf`** there is a **`revocation = relaxed|ifuri|strict`** parameter that can be set in the **`remote`** section of each connection definition. With **`ipsec.conf`** there only a general parameter **`strictcrlpolicy = yes|ifuri|no`** that can be set in the **`config setup`** section and is thus valid for all defined connections.

**Reply by sriramec** [reply]
Hi,

Looks like strictcrlpolicy is applicable only for CRLs.
if enable_ocsp is set to yes in strongswan.conf, ocsp check always works in relaxed mode.
This is the behavior in strongswan-5.6.2. 
Do I need to use higher versions of strongswan to have ocsp configured as `strict|relaxed` through `ipsec.conf`

Regards,
Sriram

**Reply by tobiasbrunner** [reply]
> if enable_ocsp is set to yes in strongswan.conf, ocsp check always works in relaxed mode.

What do you mean? Please provide logs.

**Comment by sriramec** [comment]
Hi,

This test has been done on strongswan-5.6.2
```
[root@0005B95340C0 ocspcerts]# ipsec --version
Linux strongSwan U5.6.2/K4.1.35-rt41
Institute for Internet Technologies and Applications
University of Applied Sciences Rapperswil, Switzerland
See 'ipsec --copyright' for copyright information.
```
Attaching both strongswa.conf/ipsec.conf and logs.

in strongswan.conf we have disabled CRL by doing `**enable_crl=no**`
and enabled OCSP using `**enable_ocsp=yes**`

in ipsec.conf `**strictcrlpolicy** `is set to  `**yes** `

In this case, we have not loaded signer certificate so that ocsp response signature verification fails, but the tunnel is getting established successfully.

Regards,
Sriram

[charon1.log](https://github.com/strongswan/strongswan/files/13917564/charon1.log)
[ipsec_conf.txt](https://github.com/strongswan/strongswan/files/13917565/ipsec_conf.txt)
[ipsec-up-home3.txt](https://github.com/strongswan/strongswan/files/13917566/ipsec-up-home3.txt)
[strongswan_conf.txt](https://github.com/strongswan/strongswan/files/13917567/strongswan_conf.txt)

**Reply by tobiasbrunner** [reply]
I see, yes that was fixed with 5.6.3 (cae43b890a4a23154f73c58112541de13ed9ee59). Why are you using such an old version anyway?

**Reply by sriramec** [reply]
ok Tobias. We will upgrade. 
As 5.6.2 was working just fine, we didnt upgrade.

Regards,
Sriram
